classes/_registry.py

```diff
@@ -1,32 +1,48 @@
-from typing import Callable, Dict, NoReturn, Optional
+from typing import Callable, Dict, NoReturn, Optional, Tuple
+
+from typing_extensions import Final
 
 TypeRegistry = Dict[type, Callable]
 
+#: We use this to exclude `None` as a default value for `exact_type`.
+DefaultValue: Final = type('DefaultValueType', (object,), {})
```

Comment:
Maybe use something like https://www.python.org/dev/peps/pep-0484/#id30 instead of this?

Reply:
Good suggestion! 👍 

But, I like to hide that I have a default value here. Because this is an ugly implementation detail that goes around `@overload` bug in `mypy`. So, I prefer a real `type`, not a `Enum`.